### ISSUE TITLE
Add CITATON.cff

### ISSUE THREAD

**Issue by matthewfeickert** [scikit-hep/cabinetry#492]
As @ekourlit pointed out, 

https://github.com/scikit-hep/cabinetry/blob/8f78894012b118c1225ca2f09171832de3b2e4c7/README.md?plain=1#L75

, is old as https://doi.org/10.1051/epjconf/202125103067 is the DOI that should be used now, but this should all be controlled programmatically through a `CITATION.cff` file.

**Comment by alexander-held** [comment]
Good catch! This would also be a good opportunity to make the phrasing in the readme a bit more clear along the lines of "if you find `cabinetry` useful in your research, please consider citing it: [reference]".
